saga/saga.go

```diff
@@ -1,3 +1,5 @@
+// Package saga provides a generic implementation of the Saga pattern
+// that can be used as a durability mechanism for the Scoot scheduler.
```

Comment:
Might be a good place to link to external saga docs (caitie's presentation link or otherwise).